src/compiler/binder.ts

```diff
@@ -77,6 +78,13 @@ module ts {
             return new Symbol(flags, name);
         }
 
+        function setBlockScopeContainer(node: Node, cleanLocals: boolean) {

```

Comment:
I don't understand this. Why is blockScopedContainer.locals populated? I thought it should always be empty to start.